### PR TITLE
Using options from default config when not specified in user config using DeepMerge

### PR DESCRIPTION
**Pull request by okonek**
App uses options from default config when not specified in user config. That's issue #1588. Now if you won't specify an option in your config, it will be automatically added from default config.

### CODE REVIEW

**Comment by okonek** [comment]
Anyone?

**Comment by ppot** [comment]
@albinekb I don't know if it's  a good Idea to use `deepmerge`. If we go in the way to allow only custom flag. Since this is what it is. Then in this way we need to remove the whole install process of `.hyper.js` file. And only read and assign base on available flags. For this. I would say no on this part.

File: app/config/init.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 const vm = require('vm');
+const deepmerge = require('deepmerge');
```
Comment (chabou) [inline app/config/init.js]: I prefer to use `'lodash/merge'` here. More widely used.

**Comment by okonek** [comment]
I'll change tomorrow

**Comment by chabou** [comment]
Oh I think I understood.
When I used `lodash/merge` notation, it didn't mean `lodash` or `merge`.
Add `lodash` dependency and cherry-pick `merge` method when requiring:
`const merge = require('lodash/merge');`

**Comment by chabou** [comment]
And please use `yarn`. `npm@5` is ok but our project use `yarn`.

**Comment by okonek** [comment]
Ok. :) I understand like lodash or merge

**Comment by chabou** [comment]
Can you remove `package-lock.json` files and resolve conflicts?

File: app/package.json [hunk 2]
```diff
@@ -19,6 +19,8 @@
     "electron-config": "1.0.0",
     "electron-is-dev": "0.3.0",
     "electron-squirrel-startup": "1.0.0",
+
```
Comment (chabou) [inline app/package.json]: Can you remove this extra blank line?

Comment (okonek) [reply app/package.json]: Ok :)

**Comment by okonek** [comment]
Anything else?

**Comment by chabou** [comment]
Can you remove your `package-lock.json` files please?
(Sorry for the delay)

**Comment by okonek** [comment]
Done

**Comment by chabou** [comment]
Thank you so much for your help 🙏

**Comment by okonek** [comment]
Thank You too. I learned how is it to contribute to open-source project.

**Pull request merged**